lib/lambda_ethereum_consensus/logger/console_logger.ex

```diff
@@ -9,9 +9,13 @@ defmodule ConsoleLogger do
   def format(level, message, timestamp, metadata) do
     formatted_metadata = format_metadata(metadata)
 
-    [format_level(level)] ++
-      [Logger.Formatter.format(@pattern, level, message, timestamp, [])] ++
-      [formatted_metadata] ++ ["\n"]
+    [
+      format_level(level),
+      Logger.Formatter.format(@pattern, level, message, timestamp, []),
+      formatted_metadata,
+      "\n"
+    ]
+    |> IO.iodata_to_binary()
```

Comment:
This is a way to avoid the concats with `++`. `iodata_to_binary/1` receives  iodata (i.e. a list of bytes/binaries) and create a unique binary/String without creating intermediate copies (which both `++` and `#{}` do), so it could be for code readability or performance.

Reply:
It's what Rodri says. I thought it was more readable + I wanted to make sure that if anything that wasn't exactly a string was returned from any member then this would take care of converting everything gracefully. If it was just `++` one of this conversions might fail.

My motivation wasn't performance driven, I suspected this could be causing errors in some cases, but the experiment didn't fix anything. It still works exactly as before.

Reply:
Thank you both for the explanations!!